dart_application_2/bin/task.dart

```diff
@@ -0,0 +1,212 @@
+        streetNumber: streetNumber,
+        city: city,
+        district: district,
+        state: state);
+    do {
+      exceptionFlag = false;
+      print("enter roll number");
+      rollString = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateNumber(rollString);
+     //         TODO: check if the student corresponding to this roll number already exists
+    } while (exceptionFlag);
+    
+    rollNumber = int.parse(rollString);
+
+    Set<Course> courseSet = {};
+
+    while (courseSet.length < 4) {
+      exceptionFlag = false;
+      print("chose 4 courses from 0,1,2,3,4,5");
+      courseString = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateOption(courseString, 0, 5);
+      if (exceptionFlag) continue;
+      int courseNumber = int.parse(courseString);
+      courseSet.add(Course.values[courseNumber]);
+    }
+        
+
+    Student student = new Student(
+        fullName: fullName,
+        age: age,
+        address: address,
+        rollNumber: rollNumber,
+        courseSet: courseSet);
+    return student;
+  }
+
+  /*  displays details of students based on sort parameter
+    */
+  static _displayDetailsOption(List<Student> studentList) {
+    studentList = SortItems.sortItems(studentList);
+    studentList.forEach((student) {
+      print("${student.fullName}" +
+          " ${student.rollNumber}  ${student.age})" +
+          " ${student.address.streetNumber}" +
+          "${student.address.city} ${student.address.state}" +
+          "${student.courseSet}");
+    });
+  }
+
+  /**
+   * delete details of student based on given roll
+   */
+  static List<Student> _deleteDetailsOption(List<Student> studentList) {
+    String rollNumberString;
+    int rollNumber, indexRollNumber;
+    bool exceptionFlag;
+    if (studentList.isEmpty) {
+      print("No element in the list");
+      return studentList;
+    }
+    do {
+      do {
+        exceptionFlag = false;
+        print("enter roll number to be deleted");
+        rollNumberString = stdin.readLineSync()!;
+        exceptionFlag = ValidationsUtil.validateNumber(rollNumberString);
+      } while (exceptionFlag);
+      rollNumber = int.parse(rollNumberString);
+      indexRollNumber = studentList
+          .indexWhere(((student) => student.rollNumber == rollNumber));
+    } while (indexRollNumber == -1);
+    studentList.removeAt(indexRollNumber);
+    return studentList;
+  }
+
+  /**
+   * save details from in-memory to disk
+   */
+  static _saveDetailsOption(List<Student> studentList, File file) {
+    String jsonTags = jsonEncode(studentList);
+    file.writeAsStringSync(jsonTags, mode: FileMode.write);
+  }
+
+  /**
+   * chose options before closing app:
+   * 4: save and close
+   * 5: close
+   */
+  static _exitApp(List<Student> studentList, File file) {
+    bool exceptionFlag;
+    String optionString;
+    do {
+      exceptionFlag = false;
+      print("enter 4 to save, 5 to exit");
```

Comment:
Instead take a yes or no from the user
As the user if he/she wants to save the details before exiting 
If yes -> save and exit, else just exit